From bbfec76b2e914df74c1ccf1a59d9a95df0294846 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 14 Sep 2016 14:21:34 +0200
Subject: [PATCH] removable-drives applet: don't hardcode Nemo

Use Gio to open the default file manager at home dir.
---
 .../cinnamon/applets/removable-drives@cinnamon.org/applet.js     | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js
index b3143de..b61445c 100644
--- a/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/removable-drives@cinnamon.org/applet.js
@@ -1,6 +1,7 @@
 const Lang = imports.lang;
 const St = imports.gi.St;
-const Cinnamon = imports.gi.Cinnamon;
+const GLib = imports.gi.GLib;
+const Gio = imports.gi.Gio;
 const Applet = imports.ui.applet;
 const Main = imports.ui.main;
 const PopupMenu = imports.ui.popupMenu;
@@ -63,9 +64,9 @@ MyApplet.prototype = {
 
             this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
             this.menu.addAction(_("Open file manager"), function(event) {
-                let appSystem = Cinnamon.AppSystem.get_default();
-                let app = appSystem.lookup_app('nemo.desktop');
-                app.activate_full(-1, event.get_time());
+                let homeFile = Gio.file_new_for_path(GLib.get_home_dir());
+                let homeUri = homeFile.get_uri();
+                Gio.app_info_launch_default_for_uri(homeUri, null);
             });     
             
             Main.placesManager.connect('mounts-updated', Lang.bind(this, this._update));
-- 
2.10.0

